zerver/lib/bugdown/__init__.py

```diff
@@ -1877,17 +1877,20 @@ def get_email_info(realm_id: int, emails: Set[str]) -> Dict[str, FullNameInfo]:
     }
     return dct
 
-def get_possible_mentions_info(realm_id: int, full_names: Set[str]) -> Dict[str, FullNameInfo]:
-    if not full_names:
-        return dict()
+def get_possible_mentions_info(realm_id: int, mention_texts: Set[str]) -> List[FullNameInfo]:
+    if not mention_texts:
+        return list()
 
-    # Remove the trailing part of the `user|id` mention syntax.
+    # Remove the trailing part of the `name|id` mention syntax,
+    # thus storing only full names in full_names.
+    full_names = set()
     name_re = r'(?P<full_name>.+)\|\d+$'
-    for full_name in full_names.copy():
```

Comment:
We needed to do `.copy()` because we were modifying `full_names` while iterating. Now, we do not need to do that.